sale_start_end_dates/models/sale_order.py

```diff
@@ -104,45 +111,24 @@ def end_date_change(self):
         if self.end_date:
             if self.start_date and self.start_date > self.end_date:
                 self.start_date = self.end_date
-            if self.start_date:
-                number_of_days = (
-                    fields.Date.from_string(self.end_date) -
-                    fields.Date.from_string(self.start_date)).days + 1
-                if self.number_of_days != number_of_days:
-                    self.number_of_days = number_of_days
 
     @api.onchange('start_date')
     def start_date_change(self):
         if self.start_date:
             if self.end_date and self.start_date > self.end_date:
                 self.end_date = self.start_date
-            if self.end_date:
-                number_of_days = (
-                    fields.Date.from_string(self.end_date) -
-                    fields.Date.from_string(self.start_date)).days + 1
-                if self.number_of_days != number_of_days:
-                    self.number_of_days = number_of_days
-
-    @api.onchange('number_of_days')
-    def number_of_days_change(self):
-        if self.number_of_days:
-            if self.start_date:
-                end_date_dt = fields.Date.from_string(self.start_date) +\
-                    relativedelta(days=self.number_of_days - 1)
-                end_date = fields.Date.to_string(end_date_dt)
-                if self.end_date != end_date:
-                    self.end_date = end_date
-            elif self.end_date:
-                self.start_date = fields.Date.from_string(self.end_date) -\
-                    relativedelta(days=self.number_of_days - 1)
 
     @api.onchange('product_id')
     def start_end_dates_product_id_change(self):
-        if self.order_id.default_start_date:
-            self.start_date = self.order_id.default_start_date
+        if self.product_id.must_have_dates:
+            if self.order_id.default_start_date:
```

Comment:
@alexis-via No Needed `else` 
```
self.start_date = self.end_date = False
if self.order_id.default_start_date:
    self.start_date = self.order_id.default_start_date
if self.order_id.default_end_date:
    self.end_date = self.order_id.default_end_date
```

Reply:
With the code you propose, it first writes False then writes the default date. With my code, it only writes once. That's why I think it's better to keep the else.